### PR TITLE
ENG-10506: Stop the auto-reconnect loop if you close the client while it's still trying to connect

### PR DESCRIPTION
**Pull request by jhugg**

### CODE REVIEW

File: src/frontend/org/voltdb/client/Distributer.java [hunk 1]
```diff
@@ -969,7 +969,6 @@ private void subscribeToNewNode() {
             //Subscribe to topology updates before retrieving the current topo
             //so there isn't potential for lost updates
             ProcedureInvocation spi = new ProcedureInvocation(m_sysHandle.getAndDecrement(), "@Subscribe", "TOPOLOGY");
-            final ByteBuffer buf = serializeSPI(spi);
```
Comment (jhugg) [inline src/frontend/org/voltdb/client/Distributer.java]: Sorry this got put in this commit. I removed a line that did nothing.

File: tests/frontend/org/voltdb/client/TestClientFeatures.java [hunk 2]
```diff
@@ -363,14 +364,36 @@ public void testAutoReconnect() throws Exception {
...
+        }
+        assertTrue(client.getConnectedHostList().isEmpty());
+
+        client.close();
+
+        // hunt for reconnect thread to make sure it's gone
+        Map<Thread, StackTraceElement[]> stMap = Thread.getAllStackTraces();
+        for (StackTraceElement[] st : stMap.values()) {
+            for (StackTraceElement ste : st) {
+                if (ste.getMethodName().contains("connectToOneServerWithRetry")) {
```
Comment (nshi) [inline tests/frontend/org/voltdb/client/TestClientFeatures.java]: This is kind of hackish. If the method name changes, this won't catch it.

**Comment by nshi** [comment]
Review finished.

File: tests/frontend/org/voltdb/client/TestClientFeatures.java (hunk 2 above)
Comment (jhugg) [inline tests/frontend/org/voltdb/client/TestClientFeatures.java]: Better idea?

**Pull request merged**